module_3/gift_certificates_system/repository/src/main/java/com/epam/esm/util/EsmPagination.java

```diff
@@ -0,0 +1,30 @@
+package com.epam.esm.util;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Positive;
+import java.io.Serializable;
+
+@AllArgsConstructor
+@Getter
+@Setter
+@ToString
+public class EsmPagination implements Serializable {
```

Comment:
this class is more related to service and web layer, since you have here the annotations for validation, ideally, the dao layer should not be aware of these annotation, since this layer only works with the database

Reply:
I put this class in this layer, because it is used in abstract class `FindAllDao<T>` in `repository` layer.
Thanks, I will take this into account in the future.